sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/KafkaBundle.java

```diff
@@ -204,6 +207,10 @@ public <K, V> List<MessageListener<K, V>> createMessageListener(
   public <K, V> MessageProducer<K, V> registerProducer(ProducerRegistration<K, V> registration)
       throws ConfigurationException {
 
+    if (!initialized) {
+      throw new IllegalStateException("KafkaBundle not yet initialized!");
```

Comment:
Maybe give some more hints?

```suggestion
      throw new IllegalStateException("KafkaBundle not yet initialized! registerProducer can be called earliest in run of the application.");
```

Maybe the sentence could be shorter and clearer than my suggestion.

- Same for [createConsumer](https://github.com/SDA-SE/sda-dropwizard-commons/pull/2825/files/4d885145a0325966decf2e5e4afe1ecec1e28712..32894c1a1bab7f60bb10fab9850ce1fdb6796a89#diff-786e204b7cd6a5ecc49de07ceb6a254aacf7290e9144a5b95d2c4ee6622a0207R290).
- What about the other create… methods? Some use the already available function [checkInit()](https://github.com/SDA-SE/sda-dropwizard-commons/pull/2825/files/4d885145a0325966decf2e5e4afe1ecec1e28712..32894c1a1bab7f60bb10fab9850ce1fdb6796a89#diff-786e204b7cd6a5ecc49de07ceb6a254aacf7290e9144a5b95d2c4ee6622a0207R470).

Reply:
As discussed internally, we should stay with a warning log for now, to avoid potentially breaking services. We should be stricter in the next major version.

Reply:
Did not see the `checkInit()`, i have removed my init additions and just log when listeners can not be used

Reply:
updated the checkInit() method with additional check for micometer listeners, that should not be breaking and just extend the functionality imho

```
  private void checkInit() {
    if (kafkaConfiguration == null
        || micrometerProducerListener == null
        || micrometerConsumerListener == null) {
      throw new IllegalStateException("KafkaConfiguration not yet initialized!");
    }
  }
```

Reply:
The `checkInit()` is used in `createMessageListener()` and `registerProducer()` methods which seem to be the main entry points for users of this bundle.

There are a couple of `createConsumer()`, `createProducer()` methods which are public were init checks could be added. I added the question to the [feature next major version](https://github.com/SDA-SE/sda-dropwizard-commons/discussions/2435#discussioncomment-7388886) discussion.